orderbook/src/database/trades.rs

```diff
@@ -0,0 +1,375 @@
+                    sell_token: tokens[2],
+                    buy_token: tokens[3],
+                    valid_to: 12,
+                    ..Default::default()
+                },
+            },
+        ];
+        for order in orders.iter() {
+            db.insert_order(order).await.unwrap();
+        }
+
+        let trades = [
+            Trade {
+                block_number: 2,
+                log_index: 0,
+                order_uid: order_ids[1],
+                sell_amount: BigUint::from(3u32),
+                buy_amount: BigUint::from(2u32),
+                sell_amount_before_fees: BigUint::from(1u32),
+                owner: owners[0],
+                buy_token: tokens[3],
+                sell_token: tokens[2],
+            },
+            Trade {
+                block_number: 2,
+                log_index: 1,
+                order_uid: order_ids[2],
+                sell_amount: BigUint::from(4u32),
+                buy_amount: BigUint::from(3u32),
+                sell_amount_before_fees: BigUint::from(2u32),
+                owner: owners[2],
+                buy_token: tokens[3],
+                sell_token: tokens[2],
+            },
+        ];
+
+        db.insert_events(vec![
+            (
+                EventIndex {
+                    block_number: 2,
+                    log_index: 0,
+                },
+                Event::DbTrade(DbTrade {
+                    order_uid: order_ids[1],
+                    sell_amount_including_fee: U256::from(3),
+                    buy_amount: U256::from(2),
+                    fee_amount: U256::from(2),
+                }),
+            ),
+            (
+                EventIndex {
+                    block_number: 2,
+                    log_index: 1,
+                },
+                Event::DbTrade(DbTrade {
+                    order_uid: order_ids[2],
+                    sell_amount_including_fee: U256::from(4),
+                    buy_amount: U256::from(3),
+                    fee_amount: U256::from(2),
+                }),
+            ),
+        ])
+        .await
+        .unwrap();
+
+        return (owners, order_ids, trades);
+    }
+
+    async fn assert_trades(db: &Database, filter: &TradeFilter, expected: &[Trade]) {
+        let filtered = db
+            .trades(&filter)
+            .try_collect::<HashSet<Trade>>()
+            .await
+            .unwrap();
+        let expected = expected.iter().cloned().collect::<HashSet<_>>();
+        assert_eq!(filtered, expected);
+    }
+
+    #[tokio::test]
+    #[ignore]
+    async fn postgres_filter_trades_by_owner() {
+        let db = Database::new("postgresql://").unwrap();
+        db.clear().await.unwrap();
+
+        let (owners, _order_ids, trades) = populate_dummy_trade_db(db.clone()).await;
+
+        assert_trades(
+            &db,
+            &TradeFilter {
+                owner: Some(owners[1]),
+                ..Default::default()
+            },
+            &[],
```

Comment:
I don't like about this test that the information how many order we should expect for owners[1] is located somewhere in the details of `populate_dummy_trade_db`. I think this test will be easier to maintain if the information that is required for the assert is "close" to it. 

I understand you don't want to copy the boilerplate code from the first test three times. I see two options:
1. Test the filter directly in the first test (using the actual owner and some random owner)
2. Split `populate_dummy_trade` in `add_trade(owner, orderId) -> Trade`. this way you have control over how often you call it with what inputs in the test case directly.

Reply:
Actually I'm only using populate dummy trade for the two tests that use the filter, but I could probably consolidate all three (or just the two for filters) tests into one and make the different assertions on the same data.

Reply:
I have consolidated all three tests into one that uses populate dummy trade db and makes all necessary assertions with and without filters. Looks a lot better.

Reply:
This is a very nice idea and I completely agree with the maintainability argument. Just before I rework this, I am assuming it is safe to say that the orders can be populated in advance via some method called `populate_dummy_orders`since this is really what I was trying to avoid doing over and over again.

Reply:
Actually never mind this populate dummy orders, I think we can also create and insert the order at the time of add trade in most of these tests.